org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/grammarAccess/GrammarAccessExtensions.xtend

```diff
@@ -396,8 +391,61 @@ class GrammarAccessExtensions {
 		}
 	}
 	
-	def String grammarFragmentToString(EObject ele, String prefix) {
-		return serializer.grammarFragmentToString(ele, prefix)
+	def String grammarFragmentToString(EObject object, String prefix) {
+		val node = NodeModelUtils.findActualNodeFor(object)
+		if (node === null) {
+			if (object instanceof RuleCall)
+				if (object?.rule?.name !== null)
+					return process(object.rule.name, prefix)
+			return ""
+		} else {
+			return node.text.process(prefix)
+		}
+	}
+
+	private def process(String input, String prefix) {
+		// remove leading and trailing blank lines
+		var lines = input.split('(\\r?\\n)')
+		var first = 0
+		while (isBlank(lines.get(first)))
+			first++
+		var last = lines.length - 1
+		while (isBlank(lines.get(last)))
+			last--
+		lines = lines.subList(first, last + 1)
+		// remove common whitespace (e.g. leading blanks) and add prefix to each line
+		val commonWhitespace = commonLeadingWhitespace(lines)
+		for (var n=0; n<lines.length; n++)
+			lines.set(n, prefix + lines.get(n).substring(commonWhitespace.length))
+		// generate result by joining lines
+		return lines.join(System.lineSeparator)
+	}
+
+	private def isBlank(String line) {
```

Comment:
am sure there is a util in guava or xtext this somewhere

Reply:
The only existing method that does more or less the same is org.apache.commons.lang.StringUtils#isBlank. But it is in a jar not currently in the classpath of the project. I would avoid adding a dependency just because of this tiny method.

Reply:
@szarnekow do you have an idea?

Reply:
https://github.com/eclipse/xtext-lib/issues/247

Reply:
`CharMatcher.whitespace().matchesAllOf(..)`

Reply:
I see but am not convinced that it would be an improvement.
"line.trim.empty" vs. "CharMatcher.whitespace().matchesAllOf(line)" ... for me it is not more clear and longer AND depend on third party code ... I would not change it ...